src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -136,6 +141,9 @@ private static boolean isNullOrEmptyString(String str) {
   private boolean docIdIsUrl;
   private String modeOfOperation;
 
+  private Calendar updateTimestampTimezone;
+  private DateFormat formatter;
```

Comment:
You've got a threading problem here. getDocIds and getModifiedDocIds can run concurrently (though each one separately cannot). We could 1) maintain a separate formatter for each of them (slightly tedious but not a bad option), 2) synchronize the uses, 3) put these into a ThreadLocal (boo), or 4) create an instance for each call (again, slightly tedious, but not a bad option, given that calls happen every 15 minutes at best, by default). I'm leaning toward option 1.

Reply:
Changed to have separate formatter for each (option 1).
Done.